### ISSUE TITLE
Exposing componentWillUpdate and animationEnd callbacks

### ISSUE THREAD

**Issue by mr-mig** [arve0/react-animate-on-change#9]
Here is a deal: I want to be able to run animations as part of the rendering process, but only when the state/props has changed from the previous rendering. 

I can propose two options: either exposing `componentWillUpdate` as surface API with the access to `oldProps, oldState`, or at least `onAnimationEnd` callback (so that I can abuse it from the parent component :))

WDYT?

**Comment by arve0** [comment]
So only when the animation is done? Maybe you could provide an example
which shows a proposed API?

**Comment by mr-mig** [comment]
I'm experimenting with the things right now, cannot easily formulate the need for those callbacks yet.
As soon as I will have something sensible, I will add an example.

**Comment by mr-mig** [comment]
So, basically I am trying to add some animations on top of [react-virtualized](https://github.com/bvaughn/react-virtualized) list. 

I'd prefer not to mess with list sources and make all the animations composed on top of components I have so far. 

I need an animation finish hook to notify parent component that the item can be moved/removed after the css animation is actually finished (you can have a look here [ReactCSSTransitionGroupChild.js#L84](https://github.com/facebook/react/blob/master/src/addons/transitions/ReactCSSTransitionGroupChild.js#L84))

On the other hand, it may be easier for me to fork and hack the virtualized list itself than to get all the animations composed...

Saying that, I am leaving the decision on the API completely on you :)

**Comment by arve0** [comment]
I think a finish hook sounds sensible.

Something like

```js
<Animate onAnimationEnd={() => console.log('animation done')}>
   ...
</Animate>
```

I'll have a look if I find some time adding this. A PR is also welcome.

**Comment by gregorskii** [comment]
Yes this is necessary. I have a similar use case where I want the component using the animation to track whether it has run the animation once per dispatched event from redux.

Example:

Form submitted (set flag)
Animation complete (reset flag for next form submit)
